app/services/requirements/datetime_checker.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module Requirements
+  class DatetimeChecker
```

Comment:
Could we call this something more specific, like 'SchedulingChecker'? This is the pattern we've got with the other checkers. Although the logic is a universal truth, we might get a different format of input, so I don't think we can have a single 'DatetimeChecker'.

Reply:
I've renamed to `SchedulingDatetimeChecker` just to make it clear that it's checking the datetime specifically.  There will be another process dealing with the scheduling itself.